[Yan Feng] iP

justintanyf opened the pull request:
# Duke App

> "After using Duke Task, I was able to
> accomplish so much more!" </br> - Satisfied Customer

Duke Task, a **simple** and **powerful** way to track your tasks 💯

- Ask tasks
- Delete tasks
- List tasks


 Over ~5~ 7 people have used and LOVED our app!

For Java users, here is our `main()` method

```java
public static void main(String[] args) {
     new Duke(LOCAL_STORAGE_LOCATION).run();
}
```

 ## How to run our app?

  1. Click on [this link](https://github.com/justintanyf/ip)

  2. Run the .jar file

  3. Star the fork


 ## Future Plans
- [X] Get Gud
- [ ] IPO

Commit:
Add Gradle support

justintanyf committed:
Level-1

justintanyf committed:
Level-2

justintanyf committed:
Level-3

justintanyf committed:
Level-4

justintanyf committed:
Level-5

justintanyf committed:
Level-5

justintanyf committed:
Level-6

justintanyf committed:
A-TextUiTesting

justintanyf committed:
A-Enums

justintanyf committed:
branch-Level-7

justintanyf committed:
branch-Level-8

justintanyf committed:
Merge branch 'branch-Level-7'

justintanyf committed:
Merge Level-7 and Level-8

justintanyf committed:
A-MoreOOP

justintanyf committed:
A-Packages

justintanyf committed:
Level-9

justintanyf committed:
A-CodingStandard

justintanyf committed:
A-JavaDoc

justintanyf committed:
Merge branch 'branch-A-CodingStandard'

justintanyf committed:
Level-9

justintanyf committed:
Level-9

justintanyf committed:
Merge remote-tracking branch 'origin/heads/branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/Storage.java
#	src/main/java/duke/command/DoneCommand.java

justintanyf committed:
Add JavaDocs

justintanyf committed:
A-JUnit

tenebrius1 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,33 @@
+     * @return appropriate Command
+     */
+    public static Command parse(String fullCommand) {
+        if (fullCommand.equals("bye")) {
```
Maybe you can consider using a switch statement here to make the code look cleaner?

tenebrius1 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,33 @@
+package duke;
+
+import duke.command.*;
```
Remember that we are not supposed to use wildcard imports!

tenebrius1 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,114 @@
+     *
+     * @param toDoList the current list of tasks
+     */
+    public void updateLs(ArrayList<Task> toDoList) {
```
I think that this name should be `updateList` instead of `updateLs`? We were told not to use 'texting-style' spelling [here](https://nus-cs2103-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#use-standard-words).

tenebrius1 commented on src/main/java/duke/command/AddTaskCommand.java:
```diff
@@ -0,0 +1,98 @@
+            }
+            Task newTask = new ToDo(parsedUserInput[1]);
+            addTaskToList(newTask, ui, tasks, storage);
+        } else if (parsedUserInput[0].equals("deadline")) { // Add deadline
```
Maybe this part can be split up into different methods or at least different code blocks? Just a suggestion though, to me this looks a bit cluttered but no major issues!

tenebrius1 reviewed (commented):
Your code generally looks really clean and is well documented. Except for a few parts that uses a bunch of `if-else` statements, I think that you followed the coding standards and guidelines pretty closely. Well done!

justintanyf committed:
Amend with respect to TA comments

jiarong15 commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,48 @@
+                Task deletedTask = tasks.getTasks().get(taskToDelete);
+                tasks.getTasks().remove(taskToDelete);
+                storage.updateLocalStorage(tasks.getTasks());
+                ui.reply("Noted. I've removed this duke.task: \n" + deletedTask.toString() +
```
For this part maybe can abstract to the UI class itself?

jiarong15 commented on src/main/java/duke/command/DoneCommand.java:
```diff
@@ -0,0 +1,53 @@
+            int taskDone = Integer.parseInt(parsedUserInput[1]) - 1;
+            tasks.getTasks().get(taskDone).markAsCompleted();
+            storage.updateLocalStorage(tasks.getTasks());
+            ui.reply("Nice! I've marked this duke.task as done: \n" + printDoneTask(taskDone, tasks));
```
Same as above, abstract to UI class itself? so have a method for message when you delete a task and mark a task as done :)

jiarong15 reviewed (commented):
Generally quite clean and well documented

justintanyf committed:
Merge remote-tracking branch 'origin/add-gradle-support'

justintanyf committed:
A-Gradle

justintanyf committed:
A-Gradle

justintanyf committed:
Merge tag 'A-Gradle'

weichang18 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,78 @@
+
+public class Ui {
+
+    private final String lines = "--------------------------------------------------------------------------------------------";
```
Since this is a constant variable maybe it should be named LINES instead

weichang18 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,78 @@
+
+    private final String lines = "--------------------------------------------------------------------------------------------";
+
+    /**
+     * Constructor for class Ui
+     */
+    public Ui() {
+    }
```
As the constructor did not require any variables maybe you do not have to include it in your code

weichang18 reviewed (commented):
Well done! I could not spot any more issues beside those that others have mentioned. Only had minor small issues to comment on. But, overall your code is very clean.

justintanyf committed:
Level-10

justintanyf committed:
Merge branch 'branch-Level-10'

justintanyf committed:
A-Assertions

justintanyf committed:
A-Assertions

justintanyf committed:
Removed commented-out legacy code and removed unnecessary imports

This improves readability.

Continue refactoring as more features like JavaFX is implemented.

Clearing up unnecessary code is always good when noticed as it allows everyone to ease back into understanding how the program works after a hiatus, or even just enable collaboration.

Refer to
https://nus-cs2103-ay2122s1.github.io/website/schedule/week5/topics.html#readability

justintanyf committed:
Storage.java: Added stream and parallelism functionality

The local storage is likely to contain multiple items in more realistic settings.

I have added streams and parallelism functionality to improve the speed of initiating Duke, allowing the user to start using Duke ASAP.

Streams and parallelism allows us to surpass single-core limitations, splitting the work into multiple streams to improve speed of insertion into the ArrayList of things to do.

Please refer to this for more information:
https://nus-cs2103-ay2122s1.github.io/website/schedule/week5/topics.html#w5-7-java-streams

justintanyf committed:
Storage.java: Added stream and parallelism functionality

The local storage is likely to contain multiple items in more realistic settings.

I have added streams and parallelism functionality to improve the speed of initiating Duke, allowing the user to start using Duke ASAP.

Streams and parallelism allows us to surpass single-core limitations, splitting the work into multiple streams to improve speed of insertion into the ArrayList of things to do.

Please refer to this for more information:
https://nus-cs2103-ay2122s1.github.io/website/schedule/week5/topics.html#w5-7-java-streams

justintanyf committed:
Merge pull request #1 from justintanyf/branch-A-Streams

Added stream capability

justintanyf committed:
Merged branch-A-CodeQuality to Master

justintanyf committed:
Merged branch-A-Assertions to Master

justintanyf committed:
Parser.java: Added more shortcuts to add items faster

It takes too long to type out the full word.

Now the initial of the word is accepted too.

This reduces the time needed by the user to add items to the todolist.

justintanyf committed:
Merge branch 'branch-BCD-Extension'

justintanyf committed:
Added Apollo as the title, removed the usage of non-ASCII text, and sped up processes

justintanyf committed:
Added Ui.png

justintanyf committed:
Set theme jekyll-theme-minimal

justintanyf committed:
Edited the user guide

This way the pages will display correctly

Users will better understand how to utilise Duke

justintanyf committed:
Edited the user guide

This way the pages will display correctly

Users will better understand how to utilise Duke

justintanyf committed:
Made the dialogbox responsive

This way the list will display correctly

Users will better see the list

justintanyf committed:
Solved checkstyle issues

justintanyf committed:
Solved storage routing issues

justintanyf committed:
Changed the font style to improve readability

justintanyf committed:
Final amendments

Refactored and removed unnecessary code

Improved the UI/UX

justintanyf committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	src/main/java/duke/javafx/DialogBox.java

justintanyf committed:
Final amendments

justintanyf committed:
Resolved checkstyle problems

justintanyf committed:
Made final variables for String names and Switch statements

Making it easier to find the String used and edit for the future

Making the code easier to maintain

justintanyf committed:
Solved checkstyle issues

Making the code easier to read

Making the code easier to maintain in the future
